### PR TITLE
DOC: Fixing EX01 - Added examples

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Towards https://github.com/pandas-dev/pandas/issues/37875

Notes:
On the web there are only 3 custom rolling windows: `BaseIndexer`, `FixedForwardWindowIndexer` and `VariableOffsetWindowIndexer`. But there are more of them in the [code](https://github.com/pandas-dev/pandas/blob/main/pandas/core/indexers/objects.py). Like `ExpandingIndexer` etc.. 

~~Added `# doctest: +SKIP` to the last line of `BaseIndexer` because `validate_docstrings.py` was complaining.~~

I need to do \preview here as I was having a lot of issues with the html file made with Sphinx.

I'll open a separate PR to correct the User Guide on `BaseIndexer`.

### CODE REVIEW

File: pandas/core/indexers/objects.py [hunk 1]
```diff
@@ -41,17 +41,33 @@
...
+    >>> class CustomIndexer(BaseIndexer):
+    ...     def get_window_bounds(self, num_values, min_periods, center, closed, step):
+    ...         start = np.empty(num_values, dtype=np.int64)
+    ...         end = np.empty(num_values, dtype=np.int64)
+    ...         for i in range(num_values):
+    ...             end[i] = i + self.window_size
+    ...         return start, end
+    >>> df = pd.DataFrame({"values": range(5)})
+    >>> indexer = CustomIndexer(window_size=2)
+    >>> df.rolling(indexer).sum()  # doctest: +SKIP
```
Comment (MarcoGorelli) [inline pandas/core/indexers/objects.py:59]: what was it complaining about? could you show the error please?

Comment (DeaMariaLeon) [reply pandas/core/indexers/objects.py:59]:
```(pandas-dev) *[Jul25][~/pandas-dealeon]$ ./scripts/validate_docstrings.py pandas.api.indexers.BaseIndexer
+ /Users/dealeon/mambaforge/envs/pandas-dev/bin/ninja
[1/1] Generating write_version_file with a custom command
[1]    33979 segmentation fault  ./scripts/validate_docstrings.py pandas.api.indexers.BaseIndexer```

Comment (MarcoGorelli) [reply pandas/core/indexers/objects.py:59]: you need to update `start` inside the loop too (not just `end`)

Comment (DeaMariaLeon) [reply pandas/core/indexers/objects.py:60]: You are telling me that I don't know how to copy (examples that are already made)? You are correct.

**Comment by DeaMariaLeon** [comment]
It looks like failing tests are unrelated

**Review by mroeschke — APPROVED** [review]

**Pull request merged**